client/payment-details/dispute-details/dispute-footer.tsx

```diff
@@ -0,0 +1,187 @@
+								'woocommerce-payments'
+						  ),
+					submissionDateFormatted
+				),
+				{ a: disputeDocsLinkElement }
+			);
+			break;
+		case 'lost':
+			if ( isAccepted ) {
+				// Lost accepted
+				message = createInterpolateElement(
+					sprintf(
+						/* Translators: %1$s - formatted date, %2$s – the formatted dispute fee amount, <a> - link to documentation page */
+						__(
+							'This dispute was accepted and lost on %1$s. The %2$s fee has been deducted from your account, and the disputed amount returned to the cardholder. <a>Learn more about preventing disputes</a>.',
+							'woocommerce-payments'
+						),
+						closedDateFormatted,
+						disputeFeeFormatted
+					),
+					{ a: disputeDocsLinkElement }
+				);
+			} else if ( isSubmitted ) {
+				// Lost
+				message = createInterpolateElement(
+					sprintf(
+						/* Translators: %1$s - formatted date, %2$s – the formatted dispute fee amount, <a> - link to documentation page */
+						__(
+							'This dispute was lost on %1$s. The %2$s fee has been deducted from your account, and the disputed amount returned to the cardholder. <a>Learn more about preventing disputes</a>.',
+							'woocommerce-payments'
+						),
+						closedDateFormatted,
+						disputeFeeFormatted
+					),
+					{ a: disputeDocsLinkElement }
+				);
+			} else {
+				// Lost no response
+				message = createInterpolateElement(
+					sprintf(
+						/* Translators: %1$s - formatted date, %2$s – the formatted dispute fee amount, <a> - link to documentation page */
+						__(
+							'This dispute was lost on %1$s due to non-response. The %2$s fee has been deducted from your account, and the disputed amount returned to the cardholder. <a>Learn more about preventing disputes</a>.',
+							'woocommerce-payments'
+						),
+						closedDateFormatted,
+						disputeFeeFormatted
+					),
+					{ a: disputeDocsLinkElement }
+				);
+			}
+			break;
+	}
+
+	const handleClick = () => {
+		wcpayTracks.recordEvent(
+			wcpayTracks.events.TRANSACTION_DETAILS_DISPUTE_FOOTER_BUTTON_CLICK,
+			{
+				dispute_status: dispute.status,
+				dispute_submitted: isSubmitted,
+				dispute_accepted: isAccepted,
+			}
+		);
+		if ( isSubmitted ) {
+			const challengeUrl = getAdminUrl( {
+				page: 'wc-admin',
+				path: '/payments/disputes/challenge',
+				id: dispute.id,
+			} );
+			getHistory().push( challengeUrl );
+		}
+	};
+
+	return (
+		<CardFooter
+			className={ classNames( 'transaction-details-dispute-footer', {
+				'transaction-details-dispute-footer--primary':
+					dispute.status === 'under_review',
+			} ) }
+		>
+			<Flex justify="space-between">
+				<FlexItem>{ message }</FlexItem>
+				{ isSubmitted && (
+					<FlexItem className="transaction-details-dispute-footer__actions">
+						<Button variant="secondary" onClick={ handleClick }>
+							{ buttonLabel }
+						</Button>
+					</FlexItem>
+				) }
+			</Flex>
+		</CardFooter>
```

Comment:
Similar to above comments, this feels maybe over-generalised. Is it practical to implement a self-contained hard-coded-markup `CardFooter` component for each state, and then have a parent component pick which one of those to render?

Maybe a style thing, I don't know that would be better, just my opinion reading the code – not a blocker.